pantheon-bundle/src/main/java/com/redhat/pantheon/asciidoctor/extension/UuidPreProcessor.java

```diff
@@ -0,0 +1,69 @@
+package com.redhat.pantheon.asciidoctor.extension;
+
+import org.asciidoctor.ast.Document;
+import org.asciidoctor.extension.Preprocessor;
+import org.asciidoctor.extension.PreprocessorReader;
+import com.redhat.pantheon.jcr.JcrQueryHelper;
+import org.apache.sling.api.resource.ResourceResolver;
+import java.util.Optional;
+
+import org.apache.sling.api.resource.Resource;
+import java.util.ArrayList;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class UuidPreProcessor extends Preprocessor {   // (1)
+
+    private static Resource module;
+    private static final Logger log = LoggerFactory.getLogger(UuidPreProcessor.class);
+    private static String newModulePath;
+
+    public UuidPreProcessor(Resource module) {
+        this.module = module;
+    }
+
+    @Override
+    public void process(Document document, PreprocessorReader reader) {
+        
+        List<String> lines = reader.readLines(); 
+        List<String> newLines = new ArrayList<String>();
+        String[] split;
+        String uuid, newLink;
+        
+        for (String line: lines) {
+            if(line.startsWith("xref:")){
```

Comment:
This is a good start but it needs to be updated so that it works for all of these scenarios:
- Line has an "xref:" but it's not at the start of the line.
- Author uses the "<<xrefTarget,xrefLabel>>" syntax (I'm not sure how parameters are supplied in this scenario, if it's even possible... please find out)
- Author uses a "link:" instead of an xref

Reply:
Yes Ben, 
These cases look good and there are a few more cases that crossed my mind.
I would want to incorporate as many possible in this scenario.